HousingManagementSystemApi/UseCases/RetrieveAddressesUseCase.cs

```diff
@@ -32,16 +35,23 @@ public async Task<IEnumerable<PropertyAddress>> Execute(string postcode)
 
             if (postcode == "")
                 return new List<PropertyAddress>();
-            var result = await addressesGateway.SearchByPostcode(postcode);
+            var result = await this.addressesGateway.SearchByPostcode(postcode);
             var filteredAssets = new List<PropertyAddress>();
             foreach (var property in result)
             {
-                var asset = await assetGateway.RetrieveAsset(property.Reference.ID);
-                if (assetTypes.Contains(asset.AssetType))
+                var asset = await this.assetGateway.RetrieveAsset(property.Reference.ID);
+                if (this.assetTypes.Contains(asset.AssetType))
                 {
-                    filteredAssets.Add(property);
+                    var tenureInformation = await this.tenureGateway.RetrieveTenureType(asset?.Tenure?.Id);
```

Comment:
Nitpick: Please remove `this` qualifier
```suggestion
                    var tenureInformation = await tenureGateway.RetrieveTenureType(asset?.Tenure?.Id);
```